Cakefile

```diff
@@ -277,10 +277,11 @@ runTests = (CoffeeScript) ->
   # Run every test in the `test` folder, recording failures.
   files = fs.readdirSync 'test'
 
-  # Ignore generators test file if generators are not available
+  # Ignore generator dependent test files if generators are not available
   generatorsAreAvailable = '--harmony' in process.execArgv or
     '--harmony-generators' in process.execArgv
   files.splice files.indexOf('generators.coffee'), 1 if not generatorsAreAvailable
+  files.splice files.indexOf('async.coffee'), 1 if not generatorsAreAvailable
```

Comment:
I would have made an `unless generatorsAreAvailable` block with the two `files.splice` lines in it.